packages/core/src/App/Components/Layout/Header/platform-switcher.jsx

```diff
@@ -5,79 +5,80 @@ import { withRouter } from 'react-router-dom';
 import { CSSTransition } from 'react-transition-group';
 import { Icon, Text } from '@deriv/components';
 import { getPlatformInformation, isMobile } from '@deriv/shared';
-
 import { PlatformSwitcherLoader } from './Components/Preloader/platform-switcher.jsx';
 import { PlatformDropdown } from './platform-dropdown.jsx';
 import 'Sass/app/_common/components/platform-switcher.scss';
 
-class PlatformSwitcher extends React.PureComponent {
-    constructor(props) {
-        super(props);
+const PlatformSwitcher = ({ toggleDrawer, app_routing_history, platform_config }) => {
+    const [is_open, setIsOpen] = React.useState(false);
 
-        this.state = { is_open: false };
-    }
+    const is_close_drawer_fired_ref = React.useRef(false);
 
-    toggleDrawer = () => this.setState(state => ({ is_open: !state.is_open }));
-    closeDrawer = () => {
-        this.setState({ is_open: false }, () => {
-            if (typeof this.props.toggleDrawer === 'function') {
-                this.props.toggleDrawer();
+    React.useEffect(() => {
+        if (is_close_drawer_fired_ref.current) {
+            if (typeof toggleDrawer === 'function') {
+                toggleDrawer();
             }
```

Comment:
I think we should set this back to false so that this block won't run again
```suggestion
            }
            is_close_drawer_fired_ref.current = false;
```

Reply:
yes, of course, my fault